### PR TITLE
Update to conda 24.1.2, conda-libmamba-solver 24.1.0, libmambapy 1.5.6

### PR DESCRIPTION
**Pull request by jaimergp**
### Description

Update to conda 24.1.2, conda-libmamba-solver 24.1.0, libmambapy 1.5.6

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

**Comment by jaimergp** [comment]
CI errors fixed in #60

File: src/conda_patches/0003-Return-unknown-module-in-deprecations.patch [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
 --- a/conda/deprecations.py
 +++ b/conda/deprecations.py
-@@ -284,7 +284,9 @@ class DeprecationHandler:
-             module = inspect.getmodule(frame[0])
-             return (module, module.__name__)
-         except (IndexError, AttributeError):
--            raise DeprecatedError("unable to determine the calling module") from None
-+            # raise DeprecatedError("unable to determine the calling module") from None
-+            import sys as unknown
-+            return (unknown, "<unknown>")
```
Comment (jezdez) [inline src/conda_patches/0003-Return-unknown-module-in-deprecations.patch:16]: I'm not following, why is this needed?

Comment (jaimergp) [reply src/conda_patches/0003-Return-unknown-module-in-deprecations.patch:16]: PyInstaller will fail to determine the module being deprecated because some attribute was missing during the inspection. It was introduced in https://github.com/conda/conda-standalone/pull/25. I'll comment it out to see what happens (maybe we don't need it anymore?), and worst case I can post the traceback for posterity.

**Comment by jaimergp** [comment]
Well, it works without the patch now :) Can you review @jezdez? 🙏

**Review by jezdez — APPROVED** [review]
Thank you!
